docs/architecture/adr-001.md

```diff
@@ -0,0 +1,58 @@
+# ADR 001: Disable Gravity Bridge at Genesis
+
+## Changelog
+* 12-10-2021: Initial Draft
+
+## Context
+
+The Gravity Bridge module was added for fungible asset transfers from and to the Ethereum network.
+Additionally, custom code was added for directly mapping the native tokens from Gravity Bridge
+to the Ethermint module. At the time of writing, Gravity Bridge does not have a stable release
+and the codebase contains several known issues.
+
+## Decision
+
+The Gravity Bridge module-related functionality will be hidden under the `+gravity` build tag
```

Comment:
other than that, we can also do these:
- remove the registry of the gravity hooks (this should naturally be gone with gravity module itself)
- set the `gravityKeeper` to nil in `CronosKeeper`, the only place called `gravityKeeper` is the gravity hooks, so as long as nobody access it, it should be fine
- modify `SendToEthereumHandler.Handle` to return an error immediately (so an accident call from the contract will revert)

Reply:
so keep the same module, but have a dead code with nils in it that hopefully doesn't get called anywhere and the fork of gravity as a dependency for the dead code?

Reply:
> so keep the same module, but have a dead code with nils in it that hopefully doesn't get called anywhere and the fork of gravity as a dependency for the dead code?

we might just remove `gravityKeeper` field from Cronos keeper completely, that'll be cleaner. Since it's not much code, it's not difficult to re-add them back later.

Reply:
> maybe conditional build is a good idea if we want to have gravity running a testnet

alternatively, we can have a gravity branch for that.

Reply:
[trunk-based development](https://trunkbaseddevelopment.com) generally avoids long-lived feature branches.

so, it'd likely be the opposite -- once there's a mainnet binary release, that can be forked off to a release branch, while gravity may be added back to main.

anyway, is the conclusion that it's simpler to remove the code now and add it back later? 
Besides gravity, there's also the token mapping management code -- to be removed too?

and when it's added back, will that be added to the same module (so the upgrade handler would bump the "cronos" module version) or "croons-gravity" (so the upgrade module would add the new "croons-gravity" module) ?

Reply:
> so, it'd likely be the opposite -- once there's a mainnet binary release, that can be forked off to a release branch, while gravity may be added back to main.

Yeah, that's what I thought, if we are sure to add gravity back later, then the non-gravity version is a temporary one?

> Besides gravity, there's also other gravity-related code, such as the token mapping management code -- to be removed too?

I think the token mapping feature is also useful for the IBC bridge?

> and when it's added back, will that be added to the same module (so the upgrade handler would bump the "cronos" module version) or "croons-gravity" (so the upgrade module would add the new "croons-gravity" module) ?

Currently, the state of Cronos module is just the token mapping plus several parameters. So if we keep the token mapping feature in mainnet release, there may be no migration needed (other than adding the gravity module itself) when we add the gravity feature back later.

Reply:
ok, if the token mapping is useful to other native tokens (e.g. ATOM or UST over IBC), then it may be ok to keep it -- one question here would be though: what happens if someone transfers in over IBC when there's no token mapping, and later the token mapping is added? One other "issue" is mapping the address -- e.g. as Ledger enforces different coin types, one will want to use one address "outside" EVM (on IBC) and another "inside" EVM. Would that work all right with the current token mapping logic? If not, maybe it's better to remove it for now (if one wants to use other native tokens in EVM they can still do so with some manual steps, right?)?

Reply:
> To avoid maintain multiple long-live branches, and minimize conditional compiling, I think we can:
> 
> * the gravity-related logic in the cronos module should check at runtime (for example, check the nil pointer of gravity keeper)
> * conditional compile the entry points: the app and the root CLI cmd.

We can even support build two binaries in the same repo?